bpdm-orchestrator/src/main/kotlin/org/eclipse/tractusx/bpdm/orchestrator/service/GoldenRecordTaskService.kt

```diff
@@ -95,6 +98,24 @@ class GoldenRecordTaskService(
             }
     }
 
+    @Scheduled(cron = "\${bpdm.task.timeoutCheckCron}")
+    @Synchronized
+    fun checkForTimeouts() {
+        logger.debug { "Checking for timeouts" }
+
+        taskStorage.getTasksCreatedBefore(Instant.now().minus(taskConfigProperties.taskRetentionTimeout))
+            .forEach {
+                logger.info { "Removing task ${it.taskId} after retention timeout" }
+                taskStorage.removeTask(it.taskId)
```

Comment:
I would propose to not start the retention time period with the creation time but with the instant the task has finished processing. This could be a success or error. With the pending timeout in place a task is guaranteed to end in such a final state, so the retention period could start from there

Reply:
This was my initial idea too and that is what we discussed in the refinement.

But on starting implementation I realized this poses a problem to the current API: 
Upon creating a new task the caller gets back a timeout ("The timestamp until the task is removed from the Orchestrator"). 
But this is unknown upon creation if we define the retention time only starts with the task reaching its final state.
This is what I added as a comment: #https://github.com/eclipse-tractusx/bpdm/issues/588#issuecomment-1801550090

So in my opinion we have two options:
1) We deprecate and later remove the `timeout` property in `TaskProcessingStateDto`. I can't see that it's particularly useful for the caller anyways. We already deprecated `timeout` in `TaskStepReservationResponse`.
2) We define the retention period as starting with the task creation, as implemented. In practice this doesn't make a big difference if the TaskRetentionTimeout is big compared to TaskPendingTimeout.
Maybe 1) is the cleaner way.

Reply:
Yea I would go with the first choice I think. I would expect the timout only to be the pending timeout anyway, so we could avoid sending it back to the task creator as its of limited use for him. 

Or there is also a third option:

The overall time until a task is deleted is the pending time plus retention time. So even if a task is resolved sooner than the pending time, a task will always be available the total of pending + retention time

Reply:
Yes, but with the third option we would get the same result as implemented now, just then deletionTimeout = pendingTimeout + rententionTimeout.

I'll go with option 1).

Reply:
Implemented

We should take into considering the two deprecated `timeout` properties regarding the Standard.